### PR TITLE
Remove description of combining model classes

### LINKED ISSUE #8738: DOC: Remove mention of compound class in modeling
https://github.com/astropy/astropy/pull/8736#issuecomment-494163600

### PR DESCRIPTION
**Pull request by nden**
Resolves #8738

### CODE REVIEW

**Comment by astrofrog** [comment]
It looks like the HTML failure is real:

```
Warning, treated as error:
/home/circleci/project/docs/modeling/compound-models.rst:338:Exception occurred in plotting compound-models-3
 from /home/circleci/project/docs/modeling/compound-models.rst:
Traceback (most recent call last):
  File "/home/circleci/project/venv/lib/python3.6/site-packages/matplotlib/sphinxext/plot_directive.py", line 499, in run_code
    exec(code, ns)
  File "<string>", line 10, in <module>
NameError: name 'RotatedGaussian' is not defined
```

**Comment by astrofrog** [comment]
I think the 32-bit failure might be because this needs to be rebased on master to get the latest astropy-helpers submodule version.

**Comment by nden** [comment]
@astrofrog The 32 bit failure can't be solved with a rebase. This PR was made from a fresh clone and rebased on master already. For some reason circleci runs this job from my circleci account and I think it picks up master from my fork. Any ideas how I can fix this?

**Comment by pllim** [comment]
> Any ideas how I can fix this?

@nden , theoretically, if you sync your fork's `master` with Astropy's `master`, then all will be okay again?

**Comment by nden** [comment]
@pllim I am looking for a practical solution. I stopped building it on my account and hope this will help with future PRs. Since I don't expect a documentation change to break a 32bit test please feel free to review.

**Comment by pllim** [comment]
As far as this PR is concerned, it does not look related, as it choked right at the beginning at `setup`.

File: docs/modeling/compound-models.rst [hunk 1]
```diff
@@ -86,63 +86,46 @@ As discussed in the :ref:`introduction to compound models
...
 Python with the binary operators ``+``, ``-``, ``*``, ``/``, ``**``, ``|``,
-and ``&``, each of which is discussed in the following sections.  The operands
-used in these expressions may be model *classes*, or model *instances*.  In
-other words, any object for which either ``isinstance(obj, Model)`` or
-``issubclass(obj, Model)`` is `True`.
+and ``&``, each of which is discussed in the following sections.
 
 
-.. _compound-model-classes:
+.. warning:: Creating compound models by combining classes is deprecated and will be removed in v 4.0.
```
Comment (astrofrog) [inline docs/modeling/compound-models.rst]:
```suggestion
.. warning:: Creating compound models by combining classes is deprecated and will be removed in v4.0.
```

File: docs/modeling/compound-models.rst [hunk 2]
```diff
@@ -86,63 +86,46 @@ As discussed in the :ref:`introduction to compound models
...
-    >>> isinstance(TwoGaussians, Model)
-    False
-    >>> issubclass(TwoGaussians, Model)
-    True
+    >>> two_gaussians(0.2)  # doctest: +FLOAT_CMP
+    0.9985190841886609
 
-When we inspect the variable ``TwoGaussians`` by printing its representation at
-the command prompt we can get some more information about it::
+The ``print`` command provides more information about this object::
```
Comment (astrofrog) [inline docs/modeling/compound-models.rst]:
```suggestion
The ``print`` function provides more information about this object::
```

**Comment by nden** [comment]
I guess removing the circleci build from my account fixed the issue.

**Pull request merged**